lib/internal/console/constructor.js

```diff
@@ -279,56 +279,244 @@ Console.prototype[kFormatForStderr] = function(args) {
+                    typeof v === 'object' &&
+                    !isArray(v) &&
+                    ObjectKeys(v).length > 2 ? -1 : 0;
+      const opt = {
+        depth,
+        maxArrayLength: 3,
+        ...this[kGetInspectOptions](this._stdout)
+      };
+      return util.inspect(v, opt);
+    };
+    const getIndexArray = (length) => ArrayFrom(
+      { length }, (_, i) => inspect(i));
+
+    const mapIter = isMapIterator(tabularData);
+    let isKeyValue = false;
+    let i = 0;
+    if (mapIter) {
+      const res = previewEntries(tabularData, true);
+      tabularData = res[0];
+      isKeyValue = res[1];
+    }
+
+    if (isKeyValue || isMap(tabularData)) {
+      const keys = [];
+      const values = [];
+      let length = 0;
+      if (mapIter) {
+        for (; i < tabularData.length / 2; ++i) {
+          keys.push(inspect(tabularData[i * 2]));
+          values.push(inspect(tabularData[i * 2 + 1]));
+          length++;
+        }
+      } else {
+        for (const [k, v] of tabularData) {
+          keys.push(inspect(k));
+          values.push(inspect(v));
+          length++;
+        }
+      }
+      return final([
+        iterKey, keyKey, valuesKey
+      ], [
+        getIndexArray(length),
+        keys,
+        values,
+      ]);
+    }
+
+    const setIter = isSetIterator(tabularData);
+    if (setIter)
+      tabularData = previewEntries(tabularData);
+
+    const setlike = setIter || (mapIter && !isKeyValue) || isSet(tabularData);
+    if (setlike) {
+      const values = [];
+      let length = 0;
+      for (const v of tabularData) {
+        values.push(inspect(v));
+        length++;
+      }
+      return final([setlike ? iterKey : indexKey, valuesKey], [
+        getIndexArray(length),
+        values,
+      ]);
+    }
+
+    const map = {};
+    let hasPrimitives = false;
+    const valuesKeyArray = [];
+    const indexKeyArray = ObjectKeys(tabularData);
+
+    for (; i < indexKeyArray.length; i++) {
+      const item = tabularData[indexKeyArray[i]];
+      const primitive = item === null ||
+          (typeof item !== 'function' && typeof item !== 'object');
+      if (properties === undefined && primitive) {
+        hasPrimitives = true;
+        valuesKeyArray[i] = inspect(item);
+      } else {
+        const keys = properties || ObjectKeys(item);
+        for (const key of keys) {
+          if (map[key] === undefined)
+            map[key] = [];
+          if ((primitive && properties) || !hasOwnProperty(item, key))
+            map[key][i] = '';
+          else
+            map[key][i] = item == null ? item : inspect(item[key]);
+        }
+      }
+    }
+
+    const keys = ObjectKeys(map);
+    const values = ObjectValues(map);
+    if (hasPrimitives) {
+      keys.push(valuesKey);
+      values.push(valuesKeyArray);
+    }
+    keys.unshift(indexKey);
+    values.unshift(indexKeyArray);
+
+    return final(keys, values);
+  },
 };
 
+for (const method of Reflect.ownKeys(consoleMethods))
```

Comment:
Can you put this near the end of the file and keep most of the implementation closer to where they used to be to preserve more git blame history?

Reply:
Added two new lines to align some diffs. All methods are in the same order as before, but `console.table` pretty much destroys the rest. Not sure if that is fixable.